db/db.go

```diff
@@ -6,6 +6,8 @@ import (
 
 type Storer interface {
 	ListUsers(context.Context) ([]User, error)
+	CreateNewUser(context.Context, User) (User, error)
+	CheckUserByEmail(context.Context, string) (bool, User, error)
```

Comment:
i think `CheckUserByEmail(context.Context, string) (bool, User, error)` is quite ambiguous.
if error occurred then ideal bool value should be true & 
if error not occurred then bool value should be false

CheckUserByEmail method we are using bool & err variable to denote if user is present or not, which i think is confusing 

So, instead

Rename the method
From:- `CheckUserByEmail(context.Context, string) (bool, User, error)`
To :- `GetUserByEmail(context.Context, string) (User, error)`

so in GetUserByEmail method
if err == sql.NoRows then user is not present &
if err != nil && err != sql.NoRows then db error

Reply:
Done